src/ui/state/manager.rs

```diff
@@ -199,27 +227,134 @@ impl ManagerSendState {
         }
         output_amount
     }
+
+    pub fn selected_inputs(&self) -> Vec<model::Vault> {
+        self.vaults
+            .iter()
+            .cloned()
+            .filter_map(|input| {
+                if input.selected {
+                    Some(input.vault)
+                } else {
+                    None
+                }
+            })
+            .collect()
+    }
 }
 
 impl State for ManagerSendState {
     fn update(&mut self, message: Message) -> Command<Message> {
         match message {
+            Message::SpendTransaction(res) => {
+                self.processing = false;
+                match res {
+                    Ok(tx) => {
+                        self.psbt = Some((tx.spend_tx, tx.feerate));
+                    }
+                    _ => {}
+                }
+            }
+            Message::GenerateTransaction => {
+                self.processing = true;
+                let inputs = self
+                    .selected_inputs()
+                    .iter()
+                    .map(|input| input.outpoint())
+                    .collect();
+
+                let outputs: HashMap<String, u64> = self
+                    .outputs
+                    .iter()
+                    .map(|output| (output.address.clone(), output.amount().unwrap()))
+                    .collect();
+
+                return Command::perform(
+                    get_spend_tx(self.revaultd.clone(), inputs, outputs, self.feerate),
+                    Message::SpendTransaction,
+                );
+            }
+            Message::Feerate(feerate) => {
+                if !self.processing {
+                    self.feerate = feerate;
+                    self.psbt = None;
+                }
+            }
             Message::Vaults(res) => match res {
                 Ok(vlts) => self.update_vaults(vlts),
                 Err(e) => self.warning = Some(Error::RevaultDError(e)),
             },
-            Message::Next => self.view = self.view.next(),
-            Message::Previous => self.view = self.view.previous(),
+            Message::Signed(res) => match res {
+                Ok(_) => {
+                    if let ManagerSendStep::Sign { signer, .. } = &mut self.step {
+                        // During this step state has a generated psbt
+                        // and signer has a signed psbt.
+                        self.psbt = Some((
+                            signer.signed_psbt.clone().unwrap(),
+                            self.psbt.clone().unwrap().1,
+                        ));
```

Comment:
Could you expand a bit on why the signer always has a signed PSBT and it's safe to unwrap there?

Reply:
:eyes: